Isolate VulkanHppGenerator and VideoHppGenerator in CMakeLists.txt

When adding the vulkan-hpp CMake project (e.g. through FetchContent), people may sometimes want to merely access the headers under `vulkan/` and therefore only need access to the interface library providing the correct include directory:

```cmake
FetchContent_Declare(vulkan-hpp
    GIT_REPOSITORY "https://github.com/KhronosGroup/Vulkan-Hpp.git"
    GIT_TAG "v1.4.303"
    GIT_SHALLOW ON
    GIT_SUBMODULES ""
    OVERRIDE_FIND_PACKAGE
    EXCLUDE_FROM_ALL
    SYSTEM)
FetchContent_MakeAvailable(vulkan-hpp)
```

As the submodules are not necessary for this, the VulkanHppGenerator and VideoHppGenerator targets will not be set up properly (even when not used), as the `TINYXML2_SOURCES` are missing, prompting CMake to complain about not knowing what language the target should be in:
https://github.com/KhronosGroup/Vulkan-Hpp/blob/752778463e89c27677aecaf18cf80449c12ad603/CMakeLists.txt#L366-L383

Fetching as shown will result in the following error during configuration:

```
[cmake] CMake Error at build/_deps/vulkan-hpp-src/CMakeLists.txt:367 (add_executable):
[cmake]   Cannot find source file:
[cmake] 
[cmake]     build/_deps/vulkan-hpp-src/tinyxml2/tinyxml2.cpp
[cmake] 
[cmake]   Tried extensions .c .C .c++ .cc .cpp .cxx .cu .mpp .m .M .mm .ixx .cppm
[cmake]   .ccm .cxxm .c++m .h .hh .h++ .hm .hpp .hxx .in .txx .f .F .for .f77 .f90
[cmake]   .f95 .f03 .hip .ispc
[cmake] 
[cmake] 
[cmake] CMake Error at build/_deps/vulkan-hpp-src/CMakeLists.txt:367 (add_executable):
[cmake]   No SOURCES given to target: VulkanHppGenerator
[cmake] 
[cmake] 
[cmake] CMake Error at build/_deps/vulkan-hpp-src/CMakeLists.txt:374 (add_executable):
[cmake]   No SOURCES given to target: VideoHppGenerator
[cmake] 
[cmake] 
[cmake] CMake Generate step failed.  Build files cannot be regenerated correctly.
```


It would be nice if these two targets could be hidden behind an if condition or some function, making this repo a convenient way to fetch a specific header version, rather than always generate the bindings.
The current solution is to set `SOURCE_SUBDIR "disabled"` to disable the cmake configuration for this project and adding `vulkan-hpp_SOURCE_DIR` as an include directory, which works but is not as nice as having access to an INTERFACE target, especially when wanting to use the c++20 module

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -364,6 +364,12 @@ if( VULKAN_HPP_ENABLE_CPP20_MODULES )
 	target_include_directories( VulkanHppModule PUBLIC "${CMAKE_CURRENT_SOURCE_DIR}/Vulkan-Headers/include" )
 endif()
 
+# Create Vulkan-Hpp interface target
+add_library( Vulkan-Hpp INTERFACE )
+add_library( Vulkan::Hpp ALIAS Vulkan-Hpp )
+target_include_directories( Vulkan-Hpp INTERFACE $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}> )
+
+# Build Vulkan-Hpp and Video-Hpp generators
 if ( VULKAN_HPP_GENERATOR_BUILD )
 	# The generator executable
 	add_executable( VulkanHppGenerator VulkanHppGenerator.cpp VulkanHppGenerator.hpp XMLHelper.hpp ${TINYXML2_SOURCES} ${TINYXML2_HEADERS} )

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -31,6 +31,7 @@ endif()
 # all the options for this project
 option( VULKAN_HPP_PRECOMPILE "Precompile vulkan.hpp and vulkan_raii.hpp for sample builds" ON )
 option( VULKAN_HPP_RUN_GENERATOR "Run the HPP generator" OFF )
+option( VULKAN_HPP_GENERATOR_BUILD "Build the HPP generator" ON )
 option( VULKAN_HPP_SAMPLES_BUILD "Build samples" OFF )
 option( VULKAN_HPP_TESTS_BUILD "Build tests" OFF )
 option( VULKAN_HPP_SAMPLES_BUILD_ONLY_DYNAMIC "Build only dynamic. Required in case the Vulkan SDK is not available" OFF )
@@ -363,25 +364,26 @@ if( VULKAN_HPP_ENABLE_CPP20_MODULES )
 	target_include_directories( VulkanHppModule PUBLIC "${CMAKE_CURRENT_SOURCE_DIR}/Vulkan-Headers/include" )
 endif()
 
-# The generator executable
-add_executable( VulkanHppGenerator VulkanHppGenerator.cpp VulkanHppGenerator.hpp XMLHelper.hpp ${TINYXML2_SOURCES} ${TINYXML2_HEADERS} )
-vulkan_hpp__setup_warning_level( NAME VulkanHppGenerator )
-target_compile_definitions( VulkanHppGenerator PUBLIC BASE_PATH="${CMAKE_CURRENT_SOURCE_DIR}" VK_SPEC="${vk_spec}" )
-target_include_directories( VulkanHppGenerator PRIVATE ${VULKAN_HPP_TINYXML2_SRC_DIR} )
-set_target_properties( VulkanHppGenerator PROPERTIES CXX_STANDARD 20 CXX_STANDARD_REQUIRED ON )
-if( UNIX )
-	target_link_libraries( VulkanHppGenerator PUBLIC pthread )
+if ( VULKAN_HPP_GENERATOR_BUILD )
+	# The generator executable
+	add_executable( VulkanHppGenerator VulkanHppGenerator.cpp VulkanHppGenerator.hpp XMLHelper.hpp ${TINYXML2_SOURCES} ${TINYXML2_HEADERS} )
+	vulkan_hpp__setup_warning_level( NAME VulkanHppGenerator )
+	target_compile_definitions( VulkanHppGenerator PUBLIC BASE_PATH="${CMAKE_CURRENT_SOURCE_DIR}" VK_SPEC="${vk_spec}" )
+	target_include_directories( VulkanHppGenerator PRIVATE ${VULKAN_HPP_TINYXML2_SRC_DIR} )
+	set_target_properties( VulkanHppGenerator PROPERTIES CXX_STANDARD 20 CXX_STANDARD_REQUIRED ON )
+	if( UNIX )
+		target_link_libraries( VulkanHppGenerator PUBLIC pthread )
+	endif()
+
+	# The video generator executable
+	add_executable( VideoHppGenerator VideoHppGenerator.cpp VideoHppGenerator.hpp XMLHelper.hpp ${TINYXML2_SOURCES} ${TINYXML2_HEADERS} )
+	vulkan_hpp__setup_warning_level( NAME VideoHppGenerator )
+	file( TO_NATIVE_PATH ${VulkanRegistry_DIR}/video.xml video_spec )
+	string( REPLACE "\\" "\\\\" video_spec ${video_spec} )
+	target_compile_definitions( VideoHppGenerator PUBLIC BASE_PATH="${CMAKE_CURRENT_SOURCE_DIR}" VIDEO_SPEC="${video_spec}" )
+	target_include_directories( VideoHppGenerator PRIVATE  ${VULKAN_HPP_TINYXML2_SRC_DIR} )
+	set_target_properties( VideoHppGenerator PROPERTIES CXX_STANDARD 20 CXX_STANDARD_REQUIRED ON )
 endif()
-
-# The video generator executable
-add_executable( VideoHppGenerator VideoHppGenerator.cpp VideoHppGenerator.hpp XMLHelper.hpp ${TINYXML2_SOURCES} ${TINYXML2_HEADERS} )
-vulkan_hpp__setup_warning_level( NAME VideoHppGenerator )
-file( TO_NATIVE_PATH ${VulkanRegistry_DIR}/video.xml video_spec )
-string( REPLACE "\\" "\\\\" video_spec ${video_spec} )
-target_compile_definitions( VideoHppGenerator PUBLIC BASE_PATH="${CMAKE_CURRENT_SOURCE_DIR}" VIDEO_SPEC="${video_spec}" )
-target_include_directories( VideoHppGenerator PRIVATE  ${VULKAN_HPP_TINYXML2_SRC_DIR} )
-set_target_properties( VideoHppGenerator PROPERTIES CXX_STANDARD 20 CXX_STANDARD_REQUIRED ON )
-
 
 # if the generators are to be run, add a custom commands and targets
 if( VULKAN_HPP_RUN_GENERATOR )

